app/controllers/user_controller.py

```diff
@@ -126,29 +125,13 @@ def delete():
 
 @jwt_required()
 def patch():
-    authorized_keys = ["email", "phone", "name", "password"]
+    session: Session = db.session
 
     try:
         data = request.get_json()
-        session: Session = db.session
-        token = request.headers.get("Authorization").split()[-1]
-        decoded_jwt = decode_token(token)
-        user_id = decoded_jwt.get("sub")
-        user: UserModel = UserModel.query.get(user_id)
-
-        if not user:
-            raise UserNotFound
-
-        invalid_keys = []
-
-        for key, value in data.items():
-            if key in authorized_keys:
-                setattr(user, key, value)
-            else:
-                invalid_keys.append(key)
-
-        if invalid_keys:
-            raise InvalidKeysError(authorized_keys, invalid_keys)
+        user = get_user_from_token()
+        validate_type(data)
+        validate_and_setattr(data, user)
```

Comment:
You could put these two validate functions inside a `validate_keys_values` function (or something like this) and call just once in here, also you could instantiate the `authorized keys` list in here and pass as parameter to this function.

Reply:
Done